internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -999,6 +1000,37 @@ func urlForRoute(ctx context.Context, asc ASC, routeName string) (string, error)
 	return u.String(), nil
 }
 
+// publicContainerRegistries appends mirror registries and user-specified public registries to the default list
+func publicContainerRegistries(ctx context.Context, asc ASC) string {
+	var publicRegistries string = "quay.io,registry.svc.ci.openshift.org"
+	var additionalRegistries []string
+	if asc.spec.MirrorRegistryRef != nil {
+		cm := &corev1.ConfigMap{}
+		if err := asc.rec.Client.Get(ctx, types.NamespacedName{Name: asc.spec.MirrorRegistryRef.Name, Namespace: asc.namespace}, cm); err == nil {
+			if contents, ok := cm.Data[mirrorRegistryRefRegistryConfKey]; ok {
+				if tomlTree, err := toml.Load(string(contents)); err == nil {
+					if registries, ok := tomlTree.Get("unqualified-search-registries").([]interface{}); ok {
+						for _, registry := range registries {
+							if registryStr, ok := registry.(string); ok {
+								additionalRegistries = append(additionalRegistries, registryStr)
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	if asc.spec.PublicContainerRegistries != nil {
+		additionalRegistries = append(additionalRegistries, asc.spec.PublicContainerRegistries...)
+	}
+	for _, reg := range additionalRegistries {
```

Comment:
minor comment, maybe you can use https://github.com/thoas/go-funk#funkuniq

Reply:
Great suggestion, thank you! Modified the PR to use this